File: Diagrammatize.Design.Pattern/code/python/Composite/Sample/Entry.py
import abc

class Entry(abc.ABC):
    @abc.abstractmethod
    def getName(self):
        pass

    @abc.abstractmethod
    def getSize(self):
        pass

    def printList(self):
        self.printListAux("")

    @abc.abstractmethod
    def printListAux(self, prefix):
        pass

    def __str__(self):
        return self.getName() + " (" + str(self.getSize()) + ")"

File: Diagrammatize.Design.Patterns/code/cxx/Composite/Sample/Entry.cpp
#include "Entry.hpp"
#include "FileTreatmentException.hpp"

#include <sstream>
#include <iostream>

Entry::~Entry() {
}

Entry* Entry::add(Entry* entry) {
    throw FileTreatmentException{};
}

void Entry::printList() const {
    printList("");
}

std::string Entry::toString() const {
    std::ostringstream os;
    os << getName() << " (" << getSize() << ")";
    return os.str();
}

std::ostream& operator<<(std::ostream& os, const Entry& entry) {
    os << entry.toString();
    return os;
}
